MigrationGuide.md

```diff
@@ -0,0 +1,335 @@
+# Guide for migrating to OpenAI 2.0.0-beta.1 from OpenAI 1.11.0
```

Comment:
Would it be neater to move the migration guide under a "docs" folder instead of the repo's root? 🤔

Reply:
We could. I left it besides the README and CHANGELOG because that's what I usually see in the .NET repo. We don't seem to get too many docs files other than the migration guide and a possible troubleshooting guide.